### DISCUSSION TITLE
Method Illuminate\Database\Eloquent\Collection::update does not exist.

### DISCUSSION THREAD

**Discussion by AsemAbbas** [livewire/livewire#4193] [1. Help] [answered]
![image](https://user-images.githubusercontent.com/80175112/141317330-9ffe31cc-0dc9-4273-af8b-1c006f901e52.png)

The process worked for me on other pages, but here comes this problem

![image](https://user-images.githubusercontent.com/80175112/141318298-a18b857a-6486-42ea-b4e1-aa6e50df9328.png)

Please help me 😞

**Comment by ralphjsmit** [comment, marked as answer]
Judging by the error, it looks like you're trying to update a _collection_ instead of an Eloquent model instance.

It would be the same as if you would do this:
```php
$users = User::get();

// Bad example, fails!
$users->update(['columns' => 'newValueForEveryRow']);
```

Depending on what `$this->arrested` is, you can do two things:

1. `$this->arrested` consists of **multiple items** and I want to **update every item with new values**:

    ```php
   $this->arrested->toQuery()->update($validatedData);`
   ```
2. `$this->arrested` consists of **one item** and I want to **update that single item**:
   
    ```php
   $this->arrested->first()->update($validatedData);`
   ```

**Reply by AsemAbbas** [reply]
Thank you so much its work
But I don't understand what `toQuery()` did
Can you explain that pleas

**Reply by ralphjsmit** [reply]
Happy to hear that it works!

The docs explain the `toQuery()` method like this:
> The toQuery method returns an Eloquent query builder instance containing a whereIn constraint on the collection model's primary keys:

Consider the following scenario:
```
$user1 = User::create(); // has id = 1
$user2 = User::create(); // has id = 2
$user3 = User::create(); // has id = 3

$users = User::get();

$users->toQuery()
// is the same as:
User::whereIn('id', [1, 2, 3 ]) // This means: take all the users that have an ID of 1, 2, or 3
```

Let me know if that is more clear, else I'll try to explain it differently.👍

**Reply by AsemAbbas** [reply]
oh thank you it's clear now ❤️
